### PR TITLE
SAK-41866 Switch create uniform banners

### PR DESCRIPTION
**Pull request by wyu12**
https://jira.sakaiproject.org/browse/SAK-41866?jql=text%20~%20%22banners%22

Additional PR to implement Shawn's banner styles by replacing old banner classes (messageInformation, messageError, alert-information, etc.) with standardized sak-banners.

### CODE REVIEW

**Comment by mpellicer** [comment]
Looks AWESOME AND TERRIFIC! I'm very excited about this work.

File: site-manage/site-manage-group-section-role-helper/tool/src/webapp/content/templates/CreateJoinableGroups.html [hunk 1]
```diff
@@ -102,7 +102,7 @@ <h4 rsf:id="additional-title" onclick="togglePanel( this, false ); adjustDivHeig
                             <p class="shortext">
                                 <label rsf:id="pending-groups-title" class="form-control-label">Pending Groups:</label>
                             </p>
-                            <label rsf:id="pending-groups-disclaimer" class="messageInstruction">Pending groups disclaimer</label>
+                            <label rsf:id="pending-groups-disclaimer" class="sak-banner-inf">Pending groups disclaimer</label>
```
Comment (fostersdesign) [inline site-manage/site-manage-group-section-role-helper/tool/src/webapp/content/templates/CreateJoinableGroups.html]: Missing an "o" in "info".

File: site-manage/site-manage-link-helper/src/webapp/vm/sakai_link.vm [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <div class="portletBody">
 <h3>$tlang.getString('title')</h3>
 #if ($alertMessage)
-    <div class="alertMessage">$tlang.getString("gen.alert") $alertMessage</div><div style="display:block;clear:both" ></div>
+    <div class="sak-banner-error">$tlang.getString("gen.alert") $alertMessage</div><div style="display:block;clear:both" ></div>
```
Comment (fostersdesign) [inline site-manage/site-manage-link-helper/src/webapp/vm/sakai_link.vm]: Is the `clear:both` needed?

File: site-manage/site-manage-link-helper/src/webapp/vm/sakai_link.vm [hunk 3]
```diff
@@ -31,7 +31,7 @@
 </select>
 #else
 <p>
-<div class="alertMessage">$tlang.getString('error.nosites')</div><div style="display:block;clear:both" ></div>
+<div class="sak-banner-error">$tlang.getString('error.nosites')</div><div style="display:block;clear:both" ></div>
```
Comment (fostersdesign) [inline site-manage/site-manage-link-helper/src/webapp/vm/sakai_link.vm]: Is the `clear:both` needed?

File: syllabus/syllabus-app/src/webapp/syllabus/main.jsp [hunk 4]
```diff
@@ -206,7 +206,7 @@
 					<f:verbatim>
 						<div>
 							<span id="successInfo" class="success popupMessage" style="display:none; float: left;"></span>
```
Comment (fostersdesign) [inline syllabus/syllabus-app/src/webapp/syllabus/main.jsp]: What about this .success class?

File: syllabus/syllabus-app/src/webapp/syllabus/main_edit.jsp [hunk 5]
```diff
@@ -186,7 +186,7 @@
  </script>
 <div>
 	<span id="successInfo" class="success popupMessage" style="display:none; float: left;"></span>
```
Comment (fostersdesign) [inline syllabus/syllabus-app/src/webapp/syllabus/main_edit.jsp]: Another .success missed.

File: syllabus/syllabus-app/src/webapp/syllabus/main.jsp [hunk 6]
```diff
@@ -206,7 +206,7 @@
...
-							<span id="warningInfo" class="alertMessage popupMessage" style="display:none; float: left;"></span>
+							<span id="warningInfo" class="sak-banner-error popupMessage" style="display:none; float: left;"></span>
```
Comment (fostersdesign) [inline syllabus/syllabus-app/src/webapp/syllabus/main.jsp]: Are these floats needed?

File: syllabus/syllabus-app/src/webapp/syllabus/main.jsp (hunk 4 above)
Comment (fostersdesign) [inline syllabus/syllabus-app/src/webapp/syllabus/main.jsp]: Are these floats needed?

File: syllabus/syllabus-app/src/webapp/syllabus/main_edit.jsp (hunk 5 above)
Comment (fostersdesign) [inline syllabus/syllabus-app/src/webapp/syllabus/main_edit.jsp]: Are these floats needed?

File: syllabus/syllabus-app/src/webapp/syllabus/main_edit.jsp [hunk 7]
```diff
@@ -186,7 +186,7 @@
...
-	<span id="warningInfo" class="alertMessage popupMessage" style="display:none; float: left;"></span>
+	<span id="warningInfo" class="sak-banner-error popupMessage" style="display:none; float: left;"></span>
```
Comment (fostersdesign) [inline syllabus/syllabus-app/src/webapp/syllabus/main_edit.jsp]: Are these floats needed?

File: usermembership/tool/src/webapp/usermembership/sitelist.jsp [hunk 8]
```diff
@@ -31,7 +31,7 @@
 	<h:outputText value="#{SiteListBean.initValues}"/>
 	
 	<f:subview id="allowed">
-		<h:message for="allowed" fatalClass="alertMessage" fatalStyle="margin-top: 15px;" showDetail="true"/>
+		<h:message for="allowed" fatalClass="sak-banner-error" fatalStyle="margin-top: 15px;" showDetail="true"/>
```
Comment (fostersdesign) [inline usermembership/tool/src/webapp/usermembership/sitelist.jsp]: Is this inline style needed?

File: usermembership/tool/src/webapp/usermembership/userlist.jsp [hunk 9]
```diff
@@ -22,7 +22,7 @@
 	<h:outputText value="#{UserListBean.initValues}"/>
 	
 	<f:subview id="allowed">
-		<h:message for="allowed" fatalClass="alertMessage" fatalStyle="margin-top: 15px;" showDetail="true"/>
+		<h:message for="allowed" fatalClass="sak-banner-error" fatalStyle="margin-top: 15px;" showDetail="true"/>
```
Comment (fostersdesign) [inline usermembership/tool/src/webapp/usermembership/userlist.jsp]: Is this inline style needed?

File: web/web-portlet/src/bundle/vm/edit-annotatedurl.vm [hunk 10]
```diff
@@ -8,7 +8,7 @@
 	<h3>
         $heading
 	</h3>
-	#if ($alertMessage)<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
+	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
```
Comment (fostersdesign) [inline web/web-portlet/src/bundle/vm/edit-annotatedurl.vm]: Is the .clear div needed?

File: web/web-portlet/src/bundle/vm/edit-site.vm [hunk 11]
```diff
@@ -12,7 +12,7 @@ $(document).ready(function() {
         $heading
 	</h3>
     ## TODO: Validator
-	#if ($alertMessage)<div class="alertMessage">$tlang.getString("gen.alert") $alertMessage</div><div class="clear"></div>#end
+	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $alertMessage</div><div class="clear"></div>#end
```
Comment (fostersdesign) [inline web/web-portlet/src/bundle/vm/edit-site.vm]: Is the .clear div needed?

File: web/web-portlet/src/bundle/vm/edit.vm [hunk 12]
```diff
@@ -10,7 +10,7 @@ ${includeLatestJQuery}
 		$tlang.getString("gen.custom")
 	</h3>
 	#if ($alertMessage)
-	<div class="alertMessage">
+	<div class="sak-banner-error">
 	$tlang.getString("gen.alert") $alertMessage
 	</div>
 	<div class="clear"></div>
```
Comment (fostersdesign) [inline web/web-portlet/src/bundle/vm/edit.vm]: Is the .clear div needed?

File: web/web-tool/tool/src/webapp/vm/web/chef_iframe-annotatedurl-customize.vm [hunk 13]
```diff
@@ -9,7 +9,7 @@
 		$heading 
 	</h3>
     <br /><br />
-	#if ($alertMessage)<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
+	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
```
Comment (fostersdesign) [inline web/web-tool/tool/src/webapp/vm/web/chef_iframe-annotatedurl-customize.vm]: Is the .clear div needed?

File: web/web-tool/tool/src/webapp/vm/web/chef_iframe-customize.vm [hunk 14]
```diff
@@ -8,7 +8,7 @@
 	<h3>
 		$heading
 	</h3>
-	#if ($alertMessage)<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
+	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
```
Comment (fostersdesign) [inline web/web-tool/tool/src/webapp/vm/web/chef_iframe-customize.vm]: Is the .clear div needed?

File: web/web-tool/tool/src/webapp/vm/web/chef_iframe-site-customize.vm [hunk 15]
```diff
@@ -8,7 +8,7 @@
 	<h3>
 		$heading
 	</h3>
-	#if ($alertMessage)<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
+	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div><div class="clear"></div>#end
```
Comment (fostersdesign) [inline web/web-tool/tool/src/webapp/vm/web/chef_iframe-site-customize.vm]: Is the .clear div needed?

File: admin-tools/src/webapp/vm/aliases/chef_aliases_confirm_remove.vm [hunk 16]
```diff
@@ -6,8 +6,8 @@
 	<h3>
 	$tlang.getString("alias.remove")
 	</h3>
-	#if ($alertMessage)<div class="alertMessage">$tlang.getString("alias.lert") $validator.escapeHtml($alertMessage)</div>#end
-	<div class="alertMessage">
+	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("alias.lert") $validator.escapeHtml($alertMessage)</div>#end
```
[the remark this answers is not in the document]
Comment (profmikegreene) [reply admin-tools/src/webapp/vm/aliases/chef_aliases_confirm_remove.vm]: @fostersdesign happy to more critically assess each alert for the proper type. Should that be it's own JIRA/PR?

[the remark this answers is not in the document]
Comment (fostersdesign) [reply admin-tools/src/webapp/vm/aliases/chef_aliases_confirm_remove.vm]: Let's include this work in this same Jira number and PR, as it is all related.

File: admin-tools/src/webapp/vm/authz/chef_realms_confirm_remove.vm [hunk 17]
```diff
@@ -4,9 +4,9 @@
 	<h3>
 		$tlang.getString("realm.confirm.removing")
 	</h3>
-	#if ($alertMessage)<div class="alertMessage">$tlang.getString("realm.confirm.alert") $validator.escapeHtml($alertMessage)</div>#end
+	#if ($alertMessage)<div class="sak-banner-error">$tlang.getString("realm.confirm.alert") $validator.escapeHtml($alertMessage)</div>#end
```
[the remark this answers is not in the document]
Comment (wyu12) [reply admin-tools/src/webapp/vm/authz/chef_realms_confirm_remove.vm]: Before (using the alertMessage tag)
After (using the sak-banner-warn tag)

Would the warn tag be a better fit? I agree that this wouldn't be an error but I think a removal confirmation should have a more urgent color (such as the red).

[the remark this answers is not in the document]
Comment (fostersdesign) [reply admin-tools/src/webapp/vm/authz/chef_realms_confirm_remove.vm]: Yes, I think the warn banner is a better fit. 

In this case, the banner message should probably be reworded too to ask a question:

"Warning: Are you sure you want to remove the following realms?"

File: msgcntr/messageforums-app/src/webapp/jsp/dfMessageReply.jsp [hunk 18]
```diff
@@ -214,7 +214,7 @@
 			<p style="padding:0" class="act">
```
[the remark this answers is not in the document]
Comment (wyu12) [reply msgcntr/messageforums-app/src/webapp/jsp/dfMessageReply.jsp]: Changed the tag to <div> (lines 214 and 218) and changed infoClass="success" to sak-banner-success. Do you have any ideas for the inline style on line 214?

[the remark this answers is not in the document]
Comment (fostersdesign) [reply msgcntr/messageforums-app/src/webapp/jsp/dfMessageReply.jsp]: Thanks. 

That .act class must have used to have had padding on it. (It doesn't now.) So apparently they didn't want that padding. Double check in the UI to make sure nothing else is adding padding, and if not, you can remove the inline style.

**Comment by wyu12** [comment]
Closed as this work will be submitted in smaller PRs (sub-Jiras of 41866)

**Pull request closed without merging**